apps/calendar/configure.js

```diff
@@ -0,0 +1,112 @@
+/* global module, require */
+'use strict';
+var exec = require('mz/child_process').exec;
+var fs = require('mz/fs');
+var rjs = require('requirejs');
+
+module.exports = [
+  {
+    inputs: ['build/config.json'],
+    outputs: 'js/common/presets.js',
+    rule: function() {
+      return fs.readFile('./build/config.json', 'utf8').then(function(data) {
+        return fs.writeFile('./js/common/presets.js', 'define(' + data + ');');
+      });
+    }
+  },
+
+  {
+    inputs: ['build/calendar.build.js', 'js/**/*.js'],
+    outputs: [
+      'js/bundle.js',
+      'js/caldav_worker.js',
+      'js/lazy_loaded.js',
+      'js/ext/caldav.js',
+      'js/ext/ical.js'
+    ],
+    rule: function() {
+      var configFile = './build/calendar.build.js';
+      return fs.readFile(configFile, 'utf8').then(function(data) {
+        var config = JSON.parse(data);
+        return new Promise(function(resolve) {
+          rjs.optimize(config, resolve);
+        });
+      });
+    }
+  },
+
+  {
+    inputs: ['index.html'],
+    outputs: [
```

Comment:
We cannot avoid parsing task in the configure stage such as scanning shared (build/webapp-shared.js) or locale files(build/multilocale.js / build/l10n) in html, so it must take more time while expecting configure.js.

I'd like a common parser which is able to parse everything and adopt the usecase of multi-locale, l10n...etc.